cpp/src/groupby/sort/sort_helper.cu

```diff
@@ -170,22 +161,23 @@ sort_groupby_helper::index_vector const& sort_groupby_helper::group_offsets(
                                  itr,
                                  itr + size,
                                  result.begin(),
-                                 _group_offsets->begin(),
+                                 group_offsets->begin(),
                                  thrust::identity<bool>{});
   } else {
     auto const d_key_equal = comparator.equal_to<false>(
       cudf::nullate::DYNAMIC{cudf::has_nested_nulls(_keys)}, null_equality::EQUAL);
     result_end = thrust::unique_copy(rmm::exec_policy(stream),
                                      thrust::counting_iterator<size_type>(0),
                                      thrust::counting_iterator<size_type>(size),
-                                     _group_offsets->begin(),
+                                     group_offsets->begin(),
                                      permuted_row_equality_comparator(d_key_equal, sorted_order));
   }
 
-  size_type num_groups = thrust::distance(_group_offsets->begin(), result_end);
-  _group_offsets->set_element(num_groups, size, stream);
-  _group_offsets->resize(num_groups + 1, stream);
+  size_type num_groups = thrust::distance(group_offsets->begin(), result_end);
```

Comment:
This will most likely lead to a down cast which may cause compiler warning in the future. So please use `std::size_t` or `auto const` type for `num_groups`.
```suggestion
  auto const num_groups = thrust::distance(group_offsets->begin(), result_end);
```